test/gtest/db_sync.cpp

```diff
@@ -260,29 +260,43 @@ void GetPerfDbVals(const boost::filesystem::path& filename,
     }
 }
 
-void CheckKDBObjects(const boost::filesystem::path& filename,
-                     const std::string& kernel_name,
-                     const std::string& kernel_args,
-                     bool& found)
+auto LoadKDBObjects(const boost::filesystem::path& filename)
 {
-    // clang-format off
-        auto select_query = "SELECT count(*) FROM kern_db WHERE (kernel_name = ?) AND ( kernel_args = ?)";
-    // clang-format on 
-    const std::vector<std::string> value = {kernel_name, kernel_args};
-    auto sql = SQLite{filename.string(), true};
-    auto stmt = SQLite::Statement{sql, select_query, value};
-    int count = 0;
+    std::map<std::pair<std::string, std::string>, bool> kdb_cache;
+    auto select_query = "SELECT kernel_name, kernel_args from kern_db";
+    auto sql          = SQLite{filename.string(), true};
+    auto stmt         = SQLite::Statement{sql, select_query};
+    int count         = 0;
+    std::cout << "Loading kdb entries into cache" << std::endl;
     while(true)
     {
         auto rc = stmt.Step(sql);
         if(rc == SQLITE_ROW)
-            count = stmt.ColumnInt64(0);
+        {
+            ++count;
+            const auto kernel_name = stmt.ColumnText(0);
+            const auto kernel_args = stmt.ColumnText(1);
+            kdb_cache.emplace(std::make_pair(kernel_name, kernel_args), true);
+        }
         else if(rc == SQLITE_DONE)
             break;
         else if(rc == SQLITE_ERROR || rc == SQLITE_MISUSE)
             throw std::runtime_error(sql.ErrorMessage());
+        if(count % 2000 == 0)
+            std::cout << "Loaded " << count << " entries from KDB" << std::endl;
     }
-    found = count != 0;
+
+    std::cout << "Done loading " << count << " entries from kdb file: " << filename << std::endl;
+    return kdb_cache;
+}
+
+void CheckKDBObjects(const boost::filesystem::path& filename,
+                     const std::string& kernel_name,
+                     const std::string& kernel_args,
+                     bool& found)
+{
+    static const auto kdb_cache = LoadKDBObjects(filename);
+    found = kdb_cache.find(std::make_pair(kernel_name, kernel_args)) != kdb_cache.end();
 }
```

Comment:
```suggestion
bool CheckKDBObjects(const boost::filesystem::path& filename,
                     const std::string& kernel_name,
                     const std::string& kernel_args)
{
    static const auto kdb_cache = LoadKDBObjects(filename);
    return kdb_cache.find(std::make_pair(kernel_name, kernel_args)) != kdb_cache.end();
}
```
And adjust all the callee code.

Reply:
Done!